setup_environment.py

```diff
@@ -51,47 +53,120 @@ def reward_min_cost(self, renew_cost, ff_cost):
             reward = -cost
         return reward
 
+    def get_next_state(self, action):
+        self.state[0] = action[0]
+        self.state[1] = action[1]
+        self.state[2] = (self.state[2] + 1) % 4
+        self.state[3] = get_sunlight()
+
+
+    def reward_function(self, action, state, energy_req_after_renewable, energy_req):
+        time = state[2]
+        if action[0] == 1:
+            if time ==2:
+                self.reward += 1
+            else:
+                self.reward -= 1
+
+        if action[1] == 1 and energy_req_after_renewable:
+            self.reward -= 100
+
+        if energy_req > 0:
+            self.reward -= 0
+
+    def reward_function_2(self, solar_energy_called, grid_energy_called, energy_demand):
+        negative_reward = (solar_energy_called-self.solar_energy)**2 + \
+                          (grid_energy_called-self.grid_energy)**2 + \
+                          (energy_demand)**2
+
+        positive_reward = self.solar_energy + self.battery
+
+        return positive_reward - negative_reward
+
+    def step_2(self, action, state):
+        self.reset()
+
+        time = state[2]
+        energy_demand = self.time_energy_requirement[time]
+
+        solar_energy_called = action[0]
+        grid_energy_called = action[1]
+
+        if time == 2:
+            self.solar_energy, self.battery = self.solar_producer.output(energy_demand)
+
```

Comment:
I think `self.solar_producer.output(energy_demand)` should be `self.solar_producer.output(solar_energy_called)` instead